File: 프로그래머스/2/42586. 기능개발/기능개발.py
from math import ceil

def solution(progresses, speeds):
    answer = []
    days = [ceil((100 - progresses[i]) / speeds[i]) for i in range(len(progresses))]
    count = 1
    day = days[0]

    for i in range(1, len(days)):
        if day < days[i]:
            answer.append(count)
            count = 1
            day = days[i]
        else:
            count += 1
    
    answer.append(count)

    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        ArrayDeque<Integer> deque = new ArrayDeque<>();
        List<Integer> answer = new ArrayList<>();
        
        for (int i = 0; i < speeds.length; i++) {
            deque.addLast((int)Math.ceil((100-progresses[i])/(double)speeds[i]));
        }
        
        int count;
        while (!deque.isEmpty()) {
            int s = deque.pollFirst();
            count = 1;
            while (!deque.isEmpty() && s >= deque.peekFirst()) {
                deque.pollFirst();
                count += 1;
            }
            answer.add(count);
        }
        
        return answer.stream().mapToInt(Integer::intValue).toArray();
    }
}